build/scripts/restore.ps1

```diff
@@ -0,0 +1,26 @@
+param ([string]$nugetZipUrl = $(throw "Need an URL to the NuGet zip") )
+
+$destination = ${env:UserProfile}
+$outFilePath = [IO.Path]::ChangeExtension([IO.Path]::GetTempFileName(), "zip")
+write-host "Downloading $nugetZipUrl -> $outFilePath"
+$client = new-object System.Net.WebClient
+$client.DownloadFile($nugetZipUrl, $outFilePath)
+
+# It's possible for restore to run in parallel on the test machines.  As such
+# we need to restore only new files to handle simultaneous restore scenarios.
```

Comment:
@Pilchie we removed the restore semaphore because it's an unworkable solution.  To be correct the semaphore would need to wrap the entire build step.  

IMHO: doing parallel runs on a machine is a mistake.  So long as we have global assets, like NuGet packages, we're always going to be hurting ourselves.  But right now that's how it is.